===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkcalendar.c,v
retrieving revision 1.107
retrieving revision 1.108
diff -u -r1.107 -r1.108
--- gtkcalendar.c	2006/04/16 05:01:46	1.107
+++ gtk+/gtk/gtkcalendar.c	2006/05/18 19:26:02	1.108
@@ -3014,7 +3014,7 @@
     }
   
   target = gtk_drag_dest_find_target (widget, context, NULL);
-  if (target == GDK_NONE)
+  if (target == GDK_NONE || context->suggested_action == 0)
     gdk_drag_status (context, 0, time);
   else
     {
